solutions/python-igraph/connected_components/connected_components.py

```diff
@@ -0,0 +1,33 @@
+import igraph as ig
+
+def weakly_connected_components(g : ig.Graph):
+    parent = sorted(list(range(len(g.vs))))
+
+    def find(x):
+        if x != parent[x]:
+            parent[x] = find(parent[x])
+        return parent[x]
+    
+    def union(x, y):
+        parentx, parenty = find(x), find(y)
+        if parentx != parenty:
+            parent[parentx] = parenty
+
+    for e in g.es:
+        union(e.source, e.target)
+    connected_components = dict()
```

Comment:
You can use a `defaultdict` from the `collections` module to avoid having to deal with the case when you are adding a vertex to the component for the first time.